src/fights.ts

```diff
@@ -1529,8 +1530,61 @@ const freeRunFightSources = [
       ],
     }
   ),
+  new FreeRunFight(
+    () =>
+      // eslint-disable-next-line libram/verify-constants
+      have($item`industrial fire extinguisher`) &&
+      get("_fireExtinguisherCharge") >= 10 &&
+      have($skill`Comprehensive Cartography`) &&
+      get("_monstersMapped") < 3 &&
+      fireExtinguishZones.some((zone) => zone.open() && !isBanished(zone.monster)),
+    (runSource: FreeRun) => {
+      const targets = fireExtinguishZones.filter(
+        (zone) => zone.open() && !isBanished(zone.monster)
+      );
+      const best = targets.sort((a, b) => mallPrice(b.item) - mallPrice(a.item))[0];
+      try {
+        // eslint-disable-next-line libram/verify-constants
+        const vortex = $skill`Fire Extinguisher: Polar Vortex`;
+        Macro.while_(`hasskill ${toInt(vortex)}`, Macro.skill(vortex))
+          .step(runSource.macro)
+          .setAutoAttack();
+        mapMonster(best.location, best.monster);
+      } finally {
+        setAutoAttack(0);
+      }
+    },
+    {
+      requirements: () => [
+        // eslint-disable-next-line libram/verify-constants
+        new Requirement([], { forceEquip: $items`industrial fire extinguisher` }),
+      ],
+    }
+  ),
 ];
 
+const fireExtinguishZones = [
+  {
+    location: $location`The Deep Dark Jungle`,
+    monster: $monster`smoke monster`,
+    item: $item`transdermal smoke patch`,
+    open: () => get("_spookyAirportToday") || get("spookyAirportAlways"),
+  },
+  {
+    location: $location`The Ice Hotel`,
+    monster: $monster`ice bartender`,
+    item: $item`perfect ice cube`,
+    open: () => get("_coldAirportToday") || get("coldAirportAlways"),
+  },
+
+  {
+    location: $location`The Haunted Library`,
+    monster: $monster`bookbat`,
+    item: $item`tattered scrap of paper`,
+    open: () => () => have($item`[7302]Spookyraven library key`),
```

Comment:
I don't think you need the double () => or the [7302]

Reply:
The [7302] is necessary, there are two keys of the same name.